corehq/apps/hqmedia/tasks.py

```diff
@@ -227,3 +217,53 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
         )
 
     DownloadBase.set_progress(build_application_zip, 100, 100)
+
+
+def find_missing_locale_ids_in_ccz(file_cache):
+    errors = [
+        _("Could not find {file_path} in CCZ").format(file_path)
+        for file_path in ('default/app_strings.txt', 'suite.xml') if file_path not in file_cache]
+    if errors:
+        return errors
+
+    # Each line of an app_strings.txt file is of the format "name.of.key=value of key"
+    # decode is necessary because Application._make_language_files calls .encode('utf-8')
+    app_strings_ids = {
+        line.decode("utf-8").split('=')[0]
+        for line in file_cache['default/app_strings.txt'].splitlines()
+    }
+
+    from corehq.apps.app_manager.xform import parse_xml
+    parsed = parse_xml(file_cache['suite.xml'])
+    suite_ids = {locale.get("id") for locale in parsed.iter("locale")}
+
+    return [
+        _("Locale ID {id} present in suite.xml but not in default app strings.").format(id=id)
+        for id in (suite_ids - app_strings_ids) if id
+    ]
+
+
+# Check that all media files present in media_suite.xml were added to the zip
+def check_ccz_multimedia_integrity(domain, fpath):
+    if not toggles.CAUTIOUS_MULTIMEDIA.enabled(domain):
+        return []
+
+    errors = []
+
+    with open(fpath, 'rb') as tmp:
+        with zipfile.ZipFile(tmp, "r") as z:
+            media_suites = [f for f in z.namelist() if re.search(r'\bmedia_suite.xml\b', f)]
+            if len(media_suites) != 1:
```

Comment:
I think this is impossible :fist_raised: :evergreen_tree:  but this doesn't handle more than one media suite

Reply:
Will update if we ever support multiple media suites.